src/code/audio/audio_load.c

```diff
@@ -1,161 +1,2110 @@
+
+void AudioLoad_DmaSlowCopyUnkMedium(s32 devAddr, s32 ramAddr, u8* size, s32 arg3) {
+}
+
+s32 AudioLoad_SlowLoadSeq(s32 seqId, u8* ramAddr, s8* isDone) {
+    AudioSlowLoad* slowLoad;
+    AudioTable* seqTable;
+    u32 size;
+
+    if (seqId >= gAudioContext.numSequences) {
+        *isDone = 0;
+        return -1;
+    }
+
+    seqId = AudioLoad_GetRealTableIndex(SEQUENCE_TABLE, seqId);
+    seqTable = AudioLoad_GetLoadTable(SEQUENCE_TABLE);
+    slowLoad = &gAudioContext.slowLoads[gAudioContext.slowLoadPos];
+    if (slowLoad->status == LOAD_STATUS_DONE) {
+        slowLoad->status = LOAD_STATUS_WAITING;
+    }
+
+    slowLoad->sample.sampleAddr = NULL;
+    slowLoad->isDone = isDone;
+    size = seqTable->entries[seqId].size;
+    size = ALIGN16(size);
+    slowLoad->curRamAddr = ramAddr;
+    slowLoad->status = LOAD_STATUS_START;
+    slowLoad->bytesRemaining = size;
+    slowLoad->ramAddr = ramAddr;
+    slowLoad->curDevAddr = seqTable->entries[seqId].romAddr;
+    slowLoad->medium = seqTable->entries[seqId].medium;
+    slowLoad->seqOrFontId = seqId;
+
+    if (slowLoad->medium == MEDIUM_UNK) {
+        slowLoad->unkMediumParam = seqTable->unkMediumParam;
+    }
+
+    gAudioContext.slowLoadPos ^= 1;
+    return 0;
+}
+
+void AudioLoad_InitAsyncLoads(void) {
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.asyncLoads); i++) {
+        gAudioContext.asyncLoads[i].status = 0;
+    }
+}
+
+AudioAsyncLoad* AudioLoad_StartAsyncLoadUnkMedium(s32 unkMediumParam, u32 devAddr, void* ramAddr, s32 size, s32 medium,
+                                                  s32 nChunks, OSMesgQueue* retQueue, s32 retMsg) {
+    AudioAsyncLoad* asyncLoad;
+
+    asyncLoad = AudioLoad_StartAsyncLoad(devAddr, ramAddr, size, medium, nChunks, retQueue, retMsg);
+    if (asyncLoad == NULL) {
+        return NULL;
+    }
+
+    osSendMesg(&gAudioContext.asyncLoadUnkMediumQueue, asyncLoad, OS_MESG_NOBLOCK);
+    asyncLoad->unkMediumParam = unkMediumParam;
+    return asyncLoad;
+}
+
+AudioAsyncLoad* AudioLoad_StartAsyncLoad(u32 devAddr, void* ramAddr, u32 size, s32 medium, s32 nChunks,
+                                         OSMesgQueue* retQueue, s32 retMsg) {
+    AudioAsyncLoad* asyncLoad;
+    s32 i;
+
+    for (i = 0; i < ARRAY_COUNT(gAudioContext.asyncLoads); i++) {
+        if (gAudioContext.asyncLoads[i].status == 0) {
+            asyncLoad = &gAudioContext.asyncLoads[i];
+            break;
+        }
+    }
+
+    // no more available async loads
+    if (i == ARRAY_COUNT(gAudioContext.asyncLoads)) {
+        return NULL;
+    }
+
+    asyncLoad->status = LOAD_STATUS_START;
+    asyncLoad->curDevAddr = devAddr;
+    asyncLoad->ramAddr = ramAddr;
+    asyncLoad->curRamAddr = ramAddr;
+    asyncLoad->bytesRemaining = size;
+
+    if (nChunks == 0) {
+        asyncLoad->chunkSize = 0x1000;
+    } else if (nChunks == 1) {
+        asyncLoad->chunkSize = size;
+    } else {
+        asyncLoad->chunkSize = (((s32)size / nChunks) + 0xFF) & ~0xFF;
```

Comment:
Maybe we could add a `ALIGN256` macro?

Reply:
`ALIGN256` is already in the main audio branch and used in a few other places. So yeah, it'll be good to add it here

Reply:
Marked as resolved, but don't see the ALIGN256 macro being used?